mod_reforged/hooks/skills/perks/perk_mastery_spear.nut

```diff
@@ -1,8 +1,118 @@
 ::Reforged.HooksMod.hook("scripts/skills/perks/perk_mastery_spear", function(q) {
-	q.onAfterUpdate <- function( _properties )
+	q.m.IsSpent <- true;
+	q.m.DamageMult <- 0.75;
+	q.m.Skills <- [
+		"actives.thrust",
+		"actives.prong"
+	];
+
+	q.create = @(__original) function()
 	{
+		__original();
+		this.m.Type = ::Const.SkillType.Perk | ::Const.SkillType.StatusEffect;
+		this.m.Description = ::Reforged.Mod.Tooltips.parseString("This character has mastered the art of fighting with the spear allowing him to attack for free once per [turn|Concept.Turn].");
+	}
+
+	q.isHidden = @() function()
+	{
+		return this.m.IsSpent;
+	}
+
+	q.getTooltip = @(__original) function()
+	{
+		local ret = __original();
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/special.png",
+			text = ::Reforged.Mod.Tooltips.parseString("The next [Thrust|Skill+thrust] or [Prong|Skill+prong_skill] costs " + ::MSU.Text.colorGreen("0") + " [Action Points|Concept.ActionPoints], builds " + ::MSU.Text.colorGreen("0") + " [Fatigue|Concept.Fatigue] but does " + ::MSU.Text.colorizeMult(this.m.DamageMult) + " reduced damage")
+		});
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/warning.png",
+			text = ::MSU.Text.colorRed("Will expire upon switching your weapon!")
```

Comment:
The text does not need to be all red when you already use the red warning icon. This just makes the text harder to read for little upside.

Reply:
I just followed the vanilla convention in this regard, whereby warnings are all red. And I prefer this as it makes them stand out a lot.

Reply:
I can't find a single vanilla tooltip which uses `"ui/icons/warning.png"` and then proceeds to color all or any text red